### ISSUE TITLE
Prompt when importing conflicting asset names

### ISSUE THREAD

**Issue by Samq64** [ScratchAddons/ScratchAddons#7177]
### Why this would be helpful

When importing a bunch of assets into a project the user might not want them to be renamed automatically.

### How this addon works

The addon would prompt the user to cancel/skip, replace or rename the asset with a textbox that is auto filled with what it would have been renamed to without the addon. Importing multiple files would add a checkbox to apply to all.

Basically just a file manager conflict dialog.

### Additional context

Closed PR: #7168

**Comment by Joeclinton1** [comment]
I will implement this addon today hopefully.

**Comment by Joeclinton1** [comment]
The relevant code for the renaming is:

[addCostumeAt()](https://github.com/scratchfoundation/scratch-vm/blob/develop/src/sprites/sprite.js#L85) and [addSound()](https://github.com/scratchfoundation/scratch-vm/blob/44d8d535414a70b77c1cab4aa4d2c47397d0180c/src/sprites/rendered-target.js#L527)

**Comment by mybearworld** [comment]
> The very act of replacing requires us to delete an existing costume, or alternatively to change it's costume object instead of adding a new one but either way polluting this isn't going to solve that.

`StringUtil.unusedName` is called in every context where a costume is created or changed. We wouldn't need to replace any costume objects it this were possible.

**Comment by Joeclinton1** [comment]
> > The very act of replacing requires us to delete an existing costume, or alternatively to change it's costume object instead of adding a new one but either way polluting this isn't going to solve that.
> 
> `StringUtil.unusedName` is called in every context where a costume is created or changed. We wouldn't need to replace any costume objects it this were possible.

I don't think you understand what is required here. We must replace the costume/sound object. That is just a dumb util function to get the first unused name in a list of names. It's clearly the wrong function to pollute.

**Comment by Joeclinton1** [comment]
Code is written, will begin debugging and hopefully testing tomorrow.

**Comment by mybearworld** [comment]
Am I misunderstanding what this is about? Isn't this exactly about overriding `StringUtil.unusedName`'s behavior? When a name is already used, it shouldn't fall back on adding `2` to it, but instead ask the user for a different name. Right?

**Comment by mybearworld** [comment]
> > Am I misunderstanding what this is about? Isn't this exactly about overriding `StringUtil.unusedName`'s behavior? When a name is already used, it shouldn't fall back on adding `2` to it, but instead ask the user for a different name. Right?
> 
> The OP wants to be able to skip, replace or rename the asset. Overriding `unusedName` would only allow renaming.

Ah, I see, sorry.

**Comment by Joeclinton1** [comment]
The addon implementation is getting closer to done, but it's definitely far from easy like I first thought it would.

**Comment by Joeclinton1** [comment]
I just realized I need to be polluting [vm.addCostume()](https://github.com/scratchfoundation/scratch-vm/blob/44d8d535414a70b77c1cab4aa4d2c47397d0180c/src/virtual-machine.js#L668) and [target.addSound()](https://github.com/scratchfoundation/scratch-vm/blob/44d8d535414a70b77c1cab4aa4d2c47397d0180c/src/sprites/rendered-target.js#L527) instead of the lower level things since they'll break if they think a costume/sound has been added but it actually hasn't. Also I need to handle which costume to select if no costume was added, and when to call emitProjectChanged or emitTargetsUpdate
